### PR TITLE
Single desktop UI.

### PR DESCRIPTION
**Pull request by lucent-sea**
This MR removes the Windows WPF UI and consolidates UI for all platforms into a singular Avalonia UI project.

- Removed WPF project.
- Removed WinForms from Windows projects.
  - It turns out that having `UseWindowsForms` property in the project, even if `UseWPF` is removed or set to false, will still include the WPF DLLs in the output.  So removing only WPF actually increased the binary size.  I needed to remove both.
- Refactored all the Windows services that were using APIs from WinForms.
- Renamed `IAvaloniaDispatcher` to `IUiDispatcher`.
- Removed top-level startup from desktop projects, because `Program.BuildAvaloniaApp` is required for Avalonia's designer.
- Refactored `GetVirtualScreenBounds` for Linux.
- Added `IDialogProvider`.
- Merged clipboard services from Linux/Windows into single service using Avalonia's clipboard API.
- Added custom "main window" for `IUiDispatcher` when running headless (e.g. as `IControlledApplicationLifetime`).
  - The `MainWindow` is normally only accessible via `IClassicDesktopStyleApplicationLifetime`, with no way to set your own.
- Moved "consent required" request into its own SignalR method that gets called explicitly.

### CODE REVIEW

**Review by cmbankester — APPROVED** [review]

File: Examples/Immense.RemoteControl.Examples.LinuxDesktopExample/Immense.RemoteControl.Examples.LinuxDesktopExample.csproj [hunk 1]
```diff
@@ -16,7 +16,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\Immense.RemoteControl.Desktop\Immense.RemoteControl.Desktop.csproj" />
+    <ProjectReference Include="..\..\Immense.RemoteControl.Desktop.Linux\Immense.RemoteControl.Desktop.Linux.csproj" />
+    <ProjectReference Include="..\..\Immense.RemoteControl.Desktop.UI\Immense.RemoteControl.Desktop.UI.csproj" />
```
[the remark this answers is not in the document]
Comment (lucent-sea) [reply Examples/Immense.RemoteControl.Examples.LinuxDesktopExample/Immense.RemoteControl.Examples.LinuxDesktopExample.csproj:20]: Yeah.  I found out that it's needed for the Avalonia designer to render.  It needs an executable project to reference the UI project directly.

File: Immense.RemoteControl.Desktop.UI/Services/UiDispatcher.cs [hunk 2]
```diff
@@ -103,16 +112,77 @@ public void Post(Action action, DispatcherPriority priority = default)
...
+        });
+    }
+
+    public async Task ShowDialog(Window window)
+    {
+        await Dispatcher.UIThread.InvokeAsync(async () =>
+        {
+            if (MainWindow is not null)
+            {
+                await window.ShowDialog(MainWindow);
```
[the remark this answers is not in the document]
Comment (lucent-sea) [reply Immense.RemoteControl.Desktop.UI/Services/UiDispatcher.cs:137]: Thanks.  There was one method (`ShowWindow`) that I forgot to convert to the project that uses the main window from any lifecycle style.  The other places that use `_mainWindow` are specifically for headless style (i.e. `IControlledApplicationLifetime`). 

I renamed the field to `_headlessMainWindow` so it was a little more clear.

File: Immense.RemoteControl.Desktop.UI/Services/UiDispatcher.cs [hunk 3]
```diff
@@ -103,16 +112,77 @@ public void Post(Action action, DispatcherPriority priority = default)
...
+            {
+                window.Show();
+            }
+        });
+    }
+
+    public void Shutdown()
+    {
+        _appCts.Cancel();
+        if (_currentApp?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime lifetime &&
```
[the remark this answers is not in the document]
Comment (lucent-sea) [reply Immense.RemoteControl.Desktop.UI/Services/UiDispatcher.cs]: I see what you're saying.  But in the places I'm doing the pattern matching, I have branching code paths depending on the lifetime style.  So I'd still need to add a condition (e.g. `if (ClassicLifetime is not null) { return ClassicLifetime.MainWindow?.Clipboard; }`).  In my mind, it makes sense as-is.

image

[the remark this answers is not in the document]
Comment (cmbankester) [reply Immense.RemoteControl.Desktop.UI/Services/UiDispatcher.cs]: Agreed! Thanks for the clarification

**Review by cmbankester — APPROVED** [review]

**Pull request merged**